validator/client/attest_protect.go

```diff
@@ -41,17 +47,23 @@ func (v *validator) preAttSignValidations(ctx context.Context, indexedAtt *ethpb
 	return nil
 }
 
-func (v *validator) postAttSignUpdate(ctx context.Context, indexedAtt *ethpb.IndexedAttestation, pubKey [48]byte) error {
+func (v *validator) postAttSignUpdate(ctx context.Context, indexedAtt *ethpb.IndexedAttestation, pubKey [48]byte, signingRoot [32]byte) error {
 	fmtKey := fmt.Sprintf("%#x", pubKey[:])
 	v.attesterHistoryByPubKeyLock.Lock()
+	defer v.attesterHistoryByPubKeyLock.Unlock()
 	attesterHistory, ok := v.attesterHistoryByPubKey[pubKey]
 	if ok {
-		attesterHistory = markAttestationForTargetEpoch(attesterHistory, indexedAtt.Data.Source.Epoch, indexedAtt.Data.Target.Epoch)
```

Comment:
race condition found here. if few threads were trying to vote with the same key they would all pass through the preAttSignValidations as they all read the same attesterHistory and when they get to postAttSignUpdate  there are no checks to block it